pkg/webui/console/components/downlink-form/index.js

```diff
@@ -0,0 +1,137 @@
+// Copyright © 2020 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React, { useState, useCallback } from 'react'
+import * as Yup from 'yup'
+import { defineMessages } from 'react-intl'
+
+import SubmitButton from '../../../components/submit-button'
+import RadioButton from '../../../components/radio-button'
+import Checkbox from '../../../components/checkbox'
+import Input from '../../../components/input'
+import SubmitBar from '../../../components/submit-bar'
+import toast from '../../../components/toast'
+import Form from '../../../components/form'
+import Message from '../../../lib/components/message'
+import PropTypes from '../../../lib/prop-types'
+import sharedMessages from '../../../lib/shared-messages'
+import api from '../../api'
+import { hexToBase64 } from '../../lib/bytes'
+
+const m = defineMessages({
+  insertMode: 'Insert Mode',
+  insertModeDescription: 'Messages can either replace the downlink queue or append to it',
+  replace: 'Replace',
+  push: 'Push',
+  validateFPort: 'FPort must be between 1 and 223',
+  confirmation: 'Confirmation',
+  confirmed: 'Confirmed',
+  scheduleDownlink: 'Schedule Downlink',
+  downlinkSuccess: 'Downlink was scheduled successfully',
+  payloadDescription: 'The desired payload bytes of the downlink message',
+})
+
+const validationSchema = Yup.object({
+  _mode: Yup.string()
+    .oneOf(['replace', 'push'])
+    .isRequired(),
```

Comment:
```suggestion
    .required(sharedMessages.validateRequired),
```

the whole app is broken

Reply:
Regarding the missing validation message: It's not possible to not provide a value in a radio group so 🤷‍♂️.